diff --git a/surf/config.def.h b/surf/config.def.h
index 93cfeeb..76ae818 100644
--- a/surf/config.def.h
+++ b/surf/config.def.h
@@ -194,3 +194,5 @@ static Button buttons[] = {
 	{ OnAny,        0,              9,      clicknavigate,  { .i = +1 },    1 },
 	{ OnMedia,      MODKEY,         1,      clickexternplayer, { 0 },       1 },
 };
+
+#define HOMEPAGE "https://duckduckgo.com/"
diff --git a/surf/surf.c b/surf/surf.c
index 001f776..bb33919 100644
--- a/surf/surf.c
+++ b/surf/surf.c
@@ -2153,7 +2153,11 @@ main(int argc, char *argv[])
 	if (argc > 0)
 		arg.v = argv[0];
 	else
-		arg.v = "about:blank";
+		#ifdef HOMEPAGE
+			arg.v = HOMEPAGE;
+		#else
+			arg.v = "about:blank";
+		#endif
 
 	setup();
 	c = newclient(NULL);
